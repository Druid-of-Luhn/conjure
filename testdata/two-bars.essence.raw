Spec
  { language = "Essence"
  , version = [ 2
              , 0
              ]
  , topLevelBindings = [ ( Find
                         , "x"
                         , DomainIntegerFromTo (
                             Just (
                               GenericNode
                                 Negate
                                 [ ValueInteger 9
                                 ] ) ) (
                             Just ( ValueInteger 9 ) )
                         )
                       , ( Find
                         , "y"
                         , DomainIntegerFromTo (
                             Just (
                               GenericNode
                                 Negate
                                 [ ValueInteger 9
                                 ] ) ) (
                             Just ( ValueInteger 9 ) )
                         )
                       , ( Find
                         , "s1"
                         , DomainSet
                             { size = Nothing
                             , minSize = Nothing
                             , maxSize = Nothing
                             , attrDontCare = False
                             , element = DomainIntegerFromTo (
                                           Just ( ValueInteger 0 ) ) (
                                           Just ( ValueInteger 9 ) )
                             , representation = Nothing
                             }
                         )
                       , ( Find
                         , "s2"
                         , DomainSet
                             { size = Nothing
                             , minSize = Nothing
                             , maxSize = Nothing
                             , attrDontCare = False
                             , element = DomainIntegerFromTo (
                                           Just ( ValueInteger 0 ) ) (
                                           Just ( ValueInteger 9 ) )
                             , representation = Nothing
                             }
                         )
                       ]
  , topLevelWheres = []
  , objective = Nothing
  , constraints = [ GenericNode
                      Eq
                      [ GenericNode
                          Abs
                          [ Identifier "x"
                          ]
                      , Identifier "y"
                      ]
                  , GenericNode
                      Eq
                      [ GenericNode
                          Card
                          [ GenericNode
                              Intersect
                              [ Identifier "s1"
                              , Identifier "s2"
                              ]
                          ]
                      , GenericNode
                          Minus
                          [ Identifier "y"
                          , Identifier "x"
                          ]
                      ]
                  ]
  }