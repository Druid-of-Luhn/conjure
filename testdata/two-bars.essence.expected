language Essence 2.0

find x : int(-9..9)
find y : int(-9..9)
find s1 : set of int(0..9)
find s2 : set of int(0..9)

such that
    |x| = y,
    card(s1 intersect s2) = y - x
